skbuild/setuptools_wrap.py

```diff
@@ -24,6 +24,11 @@
 except ImportError:
     from io import StringIO
 
+try:
+    from shutil import which
+except ImportError:
+    from distutils.spawn import find_executable as which
```

Comment:
Instead, you could relocated the `which` function currently present in the test helpers.
See https://github.com/scikit-build/scikit-build/blob/d5fa09fb28caddad73f7a7f141969cdda28266b4/tests/__init__.py#L272-L292